protocols/isis/server/device.go

```diff
@@ -0,0 +1,119 @@
+package server
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/bio-routing/bio-rd/config"
+	"github.com/bio-routing/bio-rd/protocols/device"
+	"github.com/bio-routing/bio-rd/protocols/isis/packet"
+	"github.com/pkg/errors"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type dev struct {
+	name               string
+	srv                *Server
+	sys                sys
+	up                 bool
+	passive            bool
+	p2p                bool
+	level2             *level
+	supportedProtocols []uint8
+	phy                *device.Device
+	done               chan struct{}
+	wg                 sync.WaitGroup
+	helloMethod        func()
+	receiverMethod     func()
+}
+
+type level struct {
+	HelloInterval uint16
+	HoldTime      uint16
+	Metric        uint32
+	neighbors     *neighbors
+}
+
+func newDev(srv *Server, ifcfg *config.ISISInterfaceConfig) *dev {
+	d := &dev{
+		name:               ifcfg.Name,
+		srv:                srv,
+		passive:            ifcfg.Passive,
+		p2p:                ifcfg.P2P,
+		supportedProtocols: []uint8{packet.NLPIDIPv4, packet.NLPIDIPv6},
+		done:               make(chan struct{}),
+	}
+
+	d.helloMethod = d.receiverRoutine
```

Comment:
was assigning the same routine to both handlers intended?

Reply:
Nope. But HelloRoutine did not exist yet. Just created it.